p2p/p2p-core/src/protocol.rs

```diff
@@ -75,62 +67,77 @@ pub enum PeerRequest {
     NewTransactions(NewTransactions),
 }
 
+#[derive(Debug, Clone)]
+pub enum PeerRequest {
+    Admin(AdminRequestMessage),
+    Protocol(ProtocolRequest),
+}
+
```

Comment:
I also changed the PeerRequest/Response enums, so they are now arte split by admin and protocol messages.

We would need to define a ProtocolRequest/Response enum anyway for the new protocol request handler so it makes sense to do it here instead of providing conversions.